eng/targets/ResolveReferences.targets

```diff
@@ -119,7 +119,7 @@
     <Error Condition="@(_InvalidReferenceToSharedFxOnlyAssembly->Count()) != 0"
            Text="Cannot reference &quot;%(_InvalidReferenceToSharedFxOnlyAssembly.Identity)&quot; directly because it is part of the shared framework and this project is not. Use &lt;FrameworkReference Include=&quot;Microsoft.AspNetCore.App&quot; /&gt; instead." />
 
-    <Error Condition="@(_InvalidReferenceToNonSharedFxAssembly->Count()) != 0"
+    <Error Condition="@(_InvalidReferenceToNonSharedFxAssembly->Count()) != 0 AND '$(TargetFramework)' == '$(DefaultNetCoreTargetFramework)'"
```

Comment:
The issue here is that there are certain netstandard2.0 dependencies in our extensions packages that were flagged by this logic. We should only be concerned with netcoreapp5.0 dependencies.

Reply:
There are two projects not multitargeted:
- Microsoft.Extensions.FileProviders.Embedded.Manifest.Task, but it builds a msbuild task assembly so it's fine
- Microsoft.Extensions.ObjectPool, it doesn't have any dependencies so it won't trigger this check. However, I still think we probably want to multitarget it and it was an oversight that it wasn't.

Reply:
Change Microsoft.Extensions.ObjectPool so that it is multi-targeted.

Reply:
Done